### DISCUSSION TITLE
Poorly named Relationship Methods

### DISCUSSION THREAD

**Discussion by BrekiTomasson** [laravel/framework#44732] [Ideas]
Whenever I'm querying a model to check the absence of a relationship, I have to slow down and remind myself what the method was called. `->whereHas('media')` is so obvious, so my first guess is always `->whereHasNot('media')` ... which obviously doesn't work. Then I remember that Laravel uses a contraction in the negated version, so I reach for `->whereHasnt('media')` ... which obviously also doesn't work. It's not until I'm on my third attempt that I realize that the right answer is the far more verbose (and far less obvious) `->whereDoesntHave('media')`

What I'm suggesting is that we follow in the footsteps of `->whereNull('email')` and `->whereNotNull('email')` and implement simple `Not`-based aliases to all the `Doesnt`-methods. We wouldn't have to rename anything, just alias them under the hood.

> Addendum: Yes, I know I can `Macroable` it all in there myself, release a package for it to save on having to redo it in all of my projects, etc, etc... but I really feel like all the `Doesnt`-methods in `Illuminate\Database\Eloquent\Concerns\QueriesRelationships` are poorly named and deserve a more (pardon the pun) ... Eloquent ... touch.

**Comment by henzeb** [comment]
The same applies for the parameter names for the relationships like HasOne, HasMany etc.
